### ISSUE TITLE
Discrete action

### ISSUE THREAD

**Issue by yinjiangjin** [shariqiqbal2810/maddpg-pytorch#6]
Hi shariqiqbal2810,  thanks a lot for sharing this MADDPG's implementation.  Following your default parameter setting, I ran the gym scenarios ( simple or simple_spread ) for many times. But all these results showed that the agents did not learn a good policy ( For example, the rewards in the simple_spread basically are converged to -3.). 

I found a difference between your implementation and OpenAI's implementation when setting the **discrete action**: 
OpenAI's implementation: #tensor([[ 0.0125, -0.0128, 0.0105, 0.1107, -0.0798]]
This implementation:  one-hot #tensor([[0., 1., 0., 0., 0.]].
It does not seem to be the root cause. Because I change this discrete action, the experimental results were still bad. 

For training a good agent, could you give me some advice?

**Comment by shariqiqbal2810** [comment]
Hi,

The difference you cite seems to be that the tensor you're looking at in the openAI implementation is taken before a gumbel-softmax sampling procedure, while the tensor you're looking at from this implementation is taken after. I would not advise changing the code to not use one-hot tensors, as it would not make sense.

To train a better agent, I would try tuning the hyperparameters. From what I remember, the default hyperparameters only work well for the tasks that I have gifs of in the README.

**Comment by xuangch** [comment]
@shariqiqbal2810 Hi. Can you provide the hyperparameters you used in the simple-spread? Thank you very much!
